haskellPackages: update stackage and hackage

maralorn opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #182019. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

lf- committed:
haskellPackages.aeson-typescript: fix the testsuite and unbreak

maralorn committed:
haskellPackages: stackage LTS 19.15 -> LTS 19.16

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

maralorn committed:
all-cabal-hashes: 2022-07-18T21:55:34Z -> 2022-07-21T21:13:45Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

maralorn committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
Merge pull request #182389 from lf-/fix-aeson-ts

haskellPackages.aeson-typescript: fix the testsuite and unbreak

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskellPackages.doctest-parallel: Disable defective doctests

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskellPackages.hls-ormolu-plugin: Disable flaky test

sternenseemann committed:
haskellPackages.terminfo: not a core pkg if cross compiling

GHC's cross build flavours disable the terminfo package, so it will
never be included if we are cross-compiling – setting it to null thus
breaks all builds depending on the package.

To fix this problem, we use the versioned attribute generated by
hackage2nix, just like we do for xhtml.

Closes #182785.

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskellPackages.regex-tdfa: Disable disfunctional tests for older ghcs

maralorn committed:
Merge pull request #182873 from sternenseemann/terminfo-cross

haskellPackages.terminfo: not a core pkg if cross compiling

sternenseemann committed:
haskellPackages.Spock-core: build with correct reroute version

Spock-core 0.14.0.1 now only supports reroute >= 0.7 because it exports
the new AltVar type (in an [PVP violation]).

We can luckily accomodate Spock-core, since all other [reverse deps] of
reroute are marked broken in nixpkgs at the moment.

[PVP violation]: https://github.com/agrafix/Spock/issues/180#issuecomment-1195328647
[reverse deps]: https://packdeps.haskellers.com/reverse/reroute

sternenseemann commented:
@robx `hasql-dynamic-statements` [broke again](https://hydra.nixos.org/build/184882950/nixlog/2), is this a pure constraints issue? Do we just need to downgrade `hasql-dynamic-statements`?

sternenseemann committed:
haskellPackages.{superbuffer,list-t,stm-containers}: reenable tests

HTF has been fixed a while ago actually, so these test suites can be
built and executed again. For stm-containers, a bounds issue persists,
though.

robx commented:
> @robx `hasql-dynamic-statements` [broke again](https://hydra.nixos.org/build/184882950/nixlog/2), is this a pure constraints issue? Do we just need to downgrade `hasql-dynamic-statements`?

It looks like we just got new `hasql` releases. `hasql-1.6` and `hasql-dynamic-statements-0.3.1.2` were uploaded to hackage on 2022-07-20, and the latter requires the former. From the error it looks like nixpkgs kept the old `hasql` version while pulling in the new `hasql-dynamic-statements`?

sternenseemann committed:
haskellPackages.hasql-dynamic-statements: downgrade to 0.3.1.1

We still ship hasql 1.5 due to stackage which is not compatible with
0.3.1.2.

sternenseemann commented:
@robx Thanks for your diagnosis, I've addressed the problem in ff7abeabe9c2b8e9ef0d47d8ef8cafc64b6d609f – we can just downgrade `hasql-dynamic-statements` until we switch Stackage version (hasql 1.6 will never make it into LTS 19 sadly).

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

sternenseemann committed:
haskellPackages.conduit-aeson: unbreak

* Work around broken Setup.hs which fails to parse

* Disable doctest-parallel suite

* Group doctest-parallel overrides and link the upstream issue for the
  failure; basically it does not work unless you are using cabal v2
  commands.
